Sources/Data Model/OptimizelyJSON.swift

```diff
@@ -0,0 +1,144 @@
+        do {
+            guard let data = payload.data(using: .utf8),
+                let jsonData = try JSONSerialization.jsonObject(with: data, options : .allowFragments) as? [String: Any] else
+            {
+                throw OptimizelyError.failedToConvertStringToDictionary
+            }
+            self.data = jsonData
+            self.payload = payload
+        } catch {
+            throw OptimizelyError.failedToConvertStringToDictionary
+        }
+    }
+    
+    init(data: [String: Any]) throws {
+        if !JSONSerialization.isValidJSONObject(data) {
+            throw OptimizelyError.invalidDictionary
+        }
+        self.data = data
+    }
+    
+    private init() {}
+    
+    // MARK: - OptimizelyJSON Implementation
+    
+    /// - Returns:  the string representation of json
+    /// - Throws: `OptimizelyError`
+    func toString() throws -> String {
+        guard let payload = self.payload else {
+            do {
+                let jsonData = try JSONSerialization.data(withJSONObject: data, options: [])
+                if let jsonString = String(data: jsonData, encoding: .utf8) {
+                    self.payload = jsonString
+                    return jsonString
+                }
+            } catch {
+            }
+            throw OptimizelyError.failedToConvertDictionaryToString
+        }
+        return payload
+    }
+    
+    /// - Returns:  the json dictionary
+    /// - Throws: `OptimizelyError`
+    func toMap() throws -> [String: Any] {
+        return data
+    }
+    
+    /// populates the decodable schema passed by the user
+    /// - Parameters:
+    ///   - jsonPath: Key path for the value.
+    ///   - schema: Decodable schema to populate.
+    /// - Throws: `OptimizelyError`
+    func getValue<T: Decodable>(jsonPath: String, schema: inout T) throws {
+        func populateDecodableSchema(value: Any) throws {
+            // Try to decode value into schema
+            if JSONSerialization.isValidJSONObject(value) {
+                do {
+                    let jsonData = try JSONSerialization.data(
+                        withJSONObject: value,
+                        options: .prettyPrinted
+                    )
+                    schema = try JSONDecoder().decode(T.self, from: jsonData)
+                    return
+                } catch {
+                }
+            }
+            // Assign directly to schema if possible
+            if let v = value as? T {
+                schema = v
+                return
+            }
+            throw OptimizelyError.failedToAssignValueToSchema
+        }
+        try getValue(jsonPath: jsonPath, schemaHandler: populateDecodableSchema(value:))
+    }
+    
+    /// populates the schema passed by the user
+    /// - Parameters:
+    ///   - jsonPath: Key path for the value.
+    ///   - schema: schema to populate.
+    /// - Throws: `OptimizelyError`
+    func getValue<T>(jsonPath: String, schema: inout T) throws {
```

Comment:
what's the role of this method?

Reply:
In case user provides schema such as `[Any]` which do not implement `Decodable` protocol, for such cases we have to overload the getValue method.